blockchain/types/tx_internal_data_fee_delegated_smart_contract_deploy.go

```diff
@@ -307,7 +308,7 @@ func (t *TxInternalDataFeeDelegatedSmartContractDeploy) IntrinsicGas(currentBloc
 		gas += params.TxGasHumanReadable
 	}
 
-	gasPayloadWithGas, err := IntrinsicGasPayload(gas, t.Payload)
+	gasPayloadWithGas, err := IntrinsicGasPayload(gas, t.Payload, t.GetRecipient() == nil, *fork.Rules(big.NewInt(int64(currentBlockNumber))))
```

Comment:
If `t.GetRecipient()` is not nil, does it mean that there's no contract deployment even though this is a contract deploy tx? What happens then?

Reply:
As I tested, contract deploy transaction type's recipient cannot have a recipient address. If to is not nil, tx is denied.
`{"jsonrpc":"2.0","id":83,"error":{"code":-32000,"message":"contract deploy transaction can't have a recipient address"}}` So, tx.To()==nil is always false. However, it is better to set `true` explicitly. Applied at https://github.com/klaytn/klaytn/pull/1888/commits/76a49dc8b8b2a241df27e05edb95f9e04eb8ecdf.